### PR TITLE
Refacto/tyta/changes in controller

### PR DESCRIPTION
**Pull request by kilianpichard**
Check if there is no problem / others changes to be done

### CODE REVIEW

File: src/controllers/projectCtrl.js [hunk 1]
```diff
@@ -37,66 +37,100 @@ exports.deleteProject = (req, res) => {
...
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if project exist
+  Project.findById(req.params.projectId)
```
Comment (remirubis) [inline src/controllers/projectCtrl.js]: Why did you replace the findByIdAndUpdate by findById and update within its function manually?

File: src/controllers/projectCtrl.js [hunk 2]
```diff
@@ -37,66 +37,100 @@ exports.deleteProject = (req, res) => {
...
-  Project.findByIdAndUpdate(
-    req.params.projectId,
-    { $pull: { users: req.body.userId } },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if project exist
+  Project.findById(req.params.projectId)
```
Comment (remirubis) [inline src/controllers/projectCtrl.js:92]: Same as before

File: src/controllers/teamCtrl.js [hunk 3]
```diff
@@ -44,82 +48,129 @@ exports.updateTeamName = (req, res) => {
...
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  //check if user not in the team
+  Team.findById(req.params.teamId)
```
Comment (remirubis) [inline src/controllers/teamCtrl.js:62]: Same as before

**Review by remirubis — COMMENTED** [review]
Some questions about your implementation

File: src/controllers/projectCtrl.js (hunk 1 above)
Comment (kilianpichard) [reply src/controllers/projectCtrl.js]: To add a test to check if user not already in project

File: src/controllers/projectCtrl.js (hunk 2 above)
Comment (kilianpichard) [reply src/controllers/projectCtrl.js:92]: Same, to check if user was in the team to send a Good error message (maybe there is a better way, I did this rapidly)

File: src/controllers/teamCtrl.js (hunk 3 above)
Comment (kilianpichard) [reply src/controllers/teamCtrl.js:62]: Same, to check if user was not already in team

File: src/controllers/projectCtrl.js [hunk 4]
```diff
@@ -37,66 +37,100 @@ exports.deleteProject = (req, res) => {
...
+              message: 'Project infos updated!',
+              newProject,
+            })
+          )
+          .catch((error) => res.status(500).json(error));
+      } else {
+        res.status(404).json({
+          message: 'Project not found!',
         });
+      }
```
Comment (remirubis) [inline src/controllers/projectCtrl.js]: This is precisely the same as before, I think? Do you see a difference between both logic?

**Review by remirubis — COMMENTED** [review]
Little question about a replacement of codes, I think you do the same thing but with more complexity.

File: src/controllers/projectCtrl.js (hunk 4 above)
Comment (kilianpichard) [reply src/controllers/projectCtrl.js]: You are wright on this function, I don't remember it was checking if project existed, maybe you changed it and I didn't notice and rebase

File: src/controllers/projectCtrl.js (hunk 1 above)
Comment (kilianpichard) [reply src/controllers/projectCtrl.js]: @remirubis

File: src/controllers/projectCtrl.js [hunk 5]
```diff
@@ -37,41 +37,27 @@ exports.deleteProject = (req, res) => {
...
 exports.addUserToProject = (req, res) => {
-  //check if project exist
+  // check if project exist
   Project.findById(req.params.projectId)
+    // eslint-disable-next-line consistent-return
     .then((project) => {
       if (project) {
         const user = project.users.find(
-          (user) => user.toString() === req.body.userId
+          (u) => u.toString() === req.body.userId
```
Comment (remirubis) [inline src/controllers/projectCtrl.js:60]: You prefer to put it in the body rather than in the url? As you want? :)

**Review by remirubis — COMMENTED** [review]
One comment about changes but it's fine now I think! @BrystoQ @FabienHP feedbacks please guys ?

File: src/controllers/projectCtrl.js (hunk 5 above)
Comment (kilianpichard) [reply src/controllers/projectCtrl.js:60]: @remirubis I forget to add the role, so now we need to put it in the body

File: src/controllers/teamCtrl.js (hunk 3 above)
Comment (kilianpichard) [reply src/controllers/teamCtrl.js:52]: The other conversations can be resolved ?

File: src/controllers/projectCtrl.js (hunk 5 above)
Comment (kilianpichard) [reply src/controllers/projectCtrl.js:60]: The other conversations can be resolved ?

File: src/controllers/projectCtrl.js [hunk 6]
```diff
@@ -37,45 +37,46 @@ exports.deleteProject = (req, res) => {
...
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  // check if project exist
+  Project.findById(req.params.projectId)
```
Comment (remirubis) [inline src/controllers/projectCtrl.js:55]: Do you try something like this : 
```js
  Project.findOneAndUpdate(
    { id: projectId, 'users.id': { $ne: userId } },
    { $addToSet: { users: { id: new ObjectId(userId) } } },
    { new: true }
  )
    .then((project) => {
      if (!project) {
        return res.status(400).json({
          message: 'User already in this project',
          data: null,
        });
      }
      res.status(200).json({
        message: 'The project has been modified correctly!',
        data: project,
      });
    })
    .catch((error) => {
      res.status(401).json({ message: 'Invalid Request!', error });
    });
```

With this part of code in projectModel.js file:

```js
projectSchema.pre('findOneAndUpdate', async function (next) {
  const project = this;

  // Test if we wants to add user in project
  if (this.getQuery()['users.id']) {
    const userId = this.getQuery()['users.id']['$ne'];
    const isUserExist = await userModel.findById(userId);
    if (!isUserExist) next('Unknown user');
  }

  next();
});
```

File: src/middleware/authMiddleware.js [hunk 7]
```diff
@@ -2,15 +2,18 @@ const jwt = require('jsonwebtoken');
 
 const jwtKey = process.env.JWT_KEY;
 
-exports.verifyToken = (req, res, next) => {
+exports.authMiddleware = (req, res, next) => {
   const token = req.headers.authorization;
 
+  console.log(token);
+
```
Comment (remirubis) [inline src/middleware/authMiddleware.js:9]: Could you delete this? :)

**Review by remirubis — APPROVED** [review]
If everything works you can merge it directly!

File: src/middleware/authMiddleware.js (hunk 7 above)
Comment (remirubis) [inline src/middleware/authMiddleware.js]: Could you remove this console log, please? :)
```suggestion
```

**Review by remirubis — CHANGES_REQUESTED** [review]
Little change

**Review by remirubis — APPROVED** [review]
LGTM 🚀

**Pull request merged**